### PR TITLE
TravisCI file fully compatible with Kinetic Kame version of ROS

### PR DESCRIPTION
**Pull request by gsilano**
The TravisCI file has been made fully compatible with the Linux distribution 16.04 and the Kinetic Kame version of ROS.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -69,39 +72,30 @@ env:
 # Install system dependencies, namely a very barebones ROS setup.
 before_install:
   - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
-  - sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
-  - sudo apt-get update -qq
-  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-catkin
+  - wget http://packages.ros.org/ros.key -O - | sudo apt-key add -
+  - sudo rm /var/lib/dpkg/lock
```
Comment (felixduvallet) [inline .travis.yml:75]: This should not be necessary in nominal conditions.

File: .travis.yml [hunk 2]
```diff
@@ -69,39 +72,30 @@ env:
 # Install system dependencies, namely a very barebones ROS setup.
 before_install:
   - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
-  - sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
-  - sudo apt-get update -qq
-  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-catkin
+  - wget http://packages.ros.org/ros.key -O - | sudo apt-key add -
```
Comment (felixduvallet) [inline .travis.yml:74]: Why deviate from the latest installation instructions:
http://wiki.ros.org/kinetic/Installation/Ubuntu#Installation.2BAC8-Ubuntu.2BAC8-Sources.Set_up_your_keys

File: .travis.yml [hunk 3]
```diff
@@ -4,6 +4,7 @@
 # Available here:
 #   - http://felixduvallet.github.io/ros-travis-integration
 #   - https://github.com/felixduvallet/ros-travis-integration
+#   - https://github.com/gsilano/ros-travis-integration
```
Comment (felixduvallet) [inline .travis.yml]: Thanks for submitting this Pull Request!
There shouldn't be a need for an additional location once this gets merged in. Please remove this line.

File: .travis.yml [hunk 4]
```diff
@@ -69,39 +72,30 @@ env:
 # Install system dependencies, namely a very barebones ROS setup.
 before_install:
   - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
-  - sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
-  - sudo apt-get update -qq
-  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-catkin
+  - wget http://packages.ros.org/ros.key -O - | sudo apt-key add -
+  - sudo rm /var/lib/dpkg/lock
+  - sudo dpkg --configure -a
```
Comment (felixduvallet) [inline .travis.yml:76]: same, I don't see why this is necessary

File: .travis.yml [hunk 5]
```diff
@@ -69,39 +72,30 @@ env:
...
+  - sudo apt-get update
+  - sudo apt-get install ros-$ROS_DISTRO-desktop-full python-catkin-pkg python-wstool python-catkin-tools
```
Comment (felixduvallet) [inline .travis.yml]: desktop-full takes a *long* time to install, so we should avoid it if possible. What packages have you found missing with the example (i.e. very basic) repository. Does the example not compile with ros-$ROS_DISTRO-catkin only?

File: .travis.yml [hunk 6]
```diff
@@ -69,39 +72,30 @@ env:
...
   - sudo rosdep init
   - rosdep update
+  - source /opt/ros/$ROS_DISTRO/setup.bash
 
 # Create a catkin workspace with the package under integration.
 install:
   - mkdir -p ~/catkin_ws/src
   - cd ~/catkin_ws/src
   - catkin_init_workspace
+  - git clone https://github.com/felixduvallet/pocketsphinx.git
```
Comment (felixduvallet) [inline .travis.yml]: This is not necessary, please remove.

File: .travis.yml [hunk 7]
```diff
@@ -69,39 +72,30 @@ env:
...
-  # source dependencies: install using wstool.
-  - cd ~/catkin_ws/src
-  - wstool init
-  - if [[ -f $ROSINSTALL_FILE ]] ; then wstool merge $ROSINSTALL_FILE ; fi
-  - wstool up
-  # package depdencies: install using rosdep.
-  - cd ~/catkin_ws
-  - rosdep install -y --from-paths src --ignore-src --rosdistro $ROS_DISTRO
+  - echo "source ~/catkin_ws/devel/setup.bash" >> ~/.bashrc
+  - source ~/.bashrc
```
Comment (felixduvallet) [inline .travis.yml]: Please keep the ROSINSTALL file and all the dependency management that goes into the before_script.

**Review by felixduvallet — CHANGES_REQUESTED** [review]
Thanks for submitting a pull request! Out of curiosity, what exactly was the minimum changes required to get kinetic working?
If you don't mind, please squash the three commits into one:
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by gsilano** [comment]
Of course, you have to change the OS and ROS versions, i.e., xenial and kinetic kame, respectively. Later, you have to add the two lines reported below immediately after the two sudo commands, in the "before_install" part of the file.

  - sudo rm /var/lib/dpkg/lock 
  - sudo dpkg --configure -a

File: .travis.yml (hunk 1 above)
Comment (gsilano) [reply .travis.yml:75]: Ok! I used the commands reported in the installation instructions of ROS Kinetic Kame, official web site.

File: .travis.yml (hunk 2 above)
Comment (gsilano) [reply .travis.yml:74]: I saw that it should be the right way by looking the instructions reported in the README file of the ROS package RotorS.

File: .travis.yml (hunk 5 above)
Comment (gsilano) [reply .travis.yml]: I used the same configuration I have been needed for my ROS package. Maybe, it is not necessary for your example. You should try.

File: .travis.yml (hunk 7 above)
Comment (gsilano) [reply .travis.yml]: Ok!!

File: .travis.yml (hunk 6 above)
Comment (gsilano) [reply .travis.yml]: Ok!!

File: .travis.yml (hunk 3 above)
Comment (gsilano) [reply .travis.yml]: Ok! I removed it!

**Comment by Ishadijcks** [comment]
Is this currently being worked on? I would love to use this .yml file for Kinetic.

**Comment by gsilano** [comment]
Currently, I am using the script with Kinetic Kame in my [CrazyS](https://github.com/gsilano/CrazyS) repository. Although it is not the best version of the script (pending the review of @felixduvallet), it works. Feel free to use it. Ah, obviously, do not forget to mention me ;-)

**Comment by felixduvallet** [comment]
@Ishadijcks I believe Kinetic support should be possible without any modifications to the .travis.yml file. I've only done a couple of experiments, let me look into it further.

**Comment by Ishadijcks** [comment]
I have another quick question. Is it possible to run roslaunch in Travis for a few seconds to see if your launchfile is configured correctly? Right now it just times out the Travis build as it gets stalled.

**Comment by felixduvallet** [comment]
@Ishadijcks I believe what you're looking for is called rostest: http://wiki.ros.org/rostest

**Comment by felixduvallet** [comment]
@Ishadijcks @gsilano I'm happy to announce kinetic is now supported as part of a build matrix that also checks indigo & jade. However, please note that xenial support on Travis is very experimental, and I've been seeing a lot of spurious build failures. 
The actual modifications to the travis installation file ended up being quite minor. See #18 for more details.

**Pull request closed without merging**